### ISSUE TITLE
Can't get `AvailabilityZone` information when `DescribeEvents`

### ISSUE THREAD

**Issue by chrislinan** [aws/aws-sdk-go-v2#1950]
### Describe the bug

Hi I am using SDK: 	"github.com/aws/aws-sdk-go-v2/service/health"

I try to use `health.NewDescribeEventsPaginator` returns the events information, but the `AvailabilityZone` attribute is `nil`

### Expected Behavior

get the `AvailabilityZone` information for each event

### Current Behavior

return value is `nil`

### Reproduction Steps

```
eventPaginator := health.NewDescribeEventsPaginator(healthClient, eventParams)
for eventPaginator.HasMorePages() {
		output, err := eventPaginator.NextPage(context.TODO())
		if err != nil {
			m.log.Fatal(err)
		}
		events = append(events, output.Events...)
	}
for _, event := range events {
    fmt.Println(event.AvailabilityZone)
} 
```

### AWS Go SDK V2 Module Versions Used

        github.com/aws/aws-sdk-go-v2 v1.17.1
	github.com/aws/aws-sdk-go-v2/config v1.17.5
	github.com/aws/aws-sdk-go-v2/credentials v1.12.18
	github.com/aws/aws-sdk-go-v2/service/cloudwatch v1.21.4
	github.com/aws/aws-sdk-go-v2/service/ec2 v1.54.4
	github.com/aws/aws-sdk-go-v2/service/elasticloadbalancing v1.14.16
	github.com/aws/aws-sdk-go-v2/service/elasticloadbalancingv2 v1.18.17
	github.com/aws/aws-sdk-go-v2/service/health v1.15.18
	github.com/aws/aws-sdk-go-v2/service/iam v1.18.17
	github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi v1.13.21
	github.com/aws/aws-sdk-go-v2/service/s3 v1.27.9
	github.com/aws/aws-sdk-go-v2/service/servicequotas v1.13.16
	github.com/aws/aws-sdk-go-v2/service/sts v1.16.17

### Compiler and Version used

go version go1.18.4 darwin/amd64

### Operating System and version

Mac OS 13.0.1 (22A400)

**Comment by chrislinan** [comment]
![image](https://user-images.githubusercontent.com/12604912/205578420-1df8381f-ec48-4fa3-b5c3-2a3ab1c8ffa1.png)

**Comment by RanVaknin** [comment]
Hi @chrislinan ,

Thanks for opening this issue. I'm not able to reproduce this issue because Im running into a `SubscriptionRequiredException`

>You must have a Business, Enterprise On-Ramp, or Enterprise Support plan from [Amazon Web Services Support](http://aws.amazon.com/premiumsupport/) to use the Health API. If you call the Health API from an Amazon Web Services account that doesn't have a Business, Enterprise On-Ramp, or Enterprise Support plan, you receive a SubscriptionRequiredException error.

Since it might take too long for me to get access to an Enterprise type tester account, I might be able to root cause this faster if you can provide some info:

Please enable response logs on your client and share them here You can do that by doing the following:
```go
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithClientLogMode(aws.LogResponseWithBody))

```
⚠️ **Warning:** Please make sure to redact your Authorization and Signature from the logs you are sharing.

That will show us the raw HTTP cycle, and if that `AvailabilityZone` is missing from the response that means the service is not returning it at all. 
If its present on the response but missing from the debugger it's probably because some serialization error. And we can further investigate this.

Thank you,
Ran~

**Comment by chrislinan** [comment]
Hi @RanVaknin 
Here is  one of the response body:
```
SDK 2022/12/06 13:45:01 DEBUG Response
HTTP/1.1 200 OK
Content-Length:4732
Content-Type:application/x-amz-json-1.1
Date: Tue, 06 Dec 2022 05:45:00 GMT
X-Amzn-Requestid:894656ae-b99b-4862-baef-482b0ecde9e5
{
   "events":[
      {
         "__type":"Event",
         "arn":"arn:aws:health:us-east-2::event/INTERNETCONNECTIVITY/AWS_INTERNETCONNECTIVITY_OPERATIONAL_ISSUE/AWS_INTERNETCONNECTIVITY_OPERATIONAL_ISSUE_7F7D7_53AC999D3F1",
         "endTime":1.670274391E9,
         "eventScopeCode":"PUBLIC",
         "eventTypeCategory":"issue",
         "eventTypeCode":"AWS_INTERNETCONNECTIVITY_OPERATIONAL_ISSUE",
         "lastUpdatedTime":1.670274391692E9,
         "region":"us-east-2",
         "service":"INTERNETCONNECTIVITY",
         "startTime":1.67027196E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED_a0fd34e3-c0ed-4772-872a-63751bca2fd0",
         "endTime":1.670249469339E9,
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"scheduledChange",
         "eventTypeCode":"AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED",
         "lastUpdatedTime":1.670249473372E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.670249469339E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED_8f573a79-85c2-47bd-850f-fe3fc26fba0e",
         "endTime":1.670189307985E9,
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"scheduledChange",
         "eventTypeCode":"AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED",
         "lastUpdatedTime":1.670189320896E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.670189307985E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:us-east-1::event/QUICKSIGHT/AWS_QUICKSIGHT_OPERATIONAL_ISSUE/AWS_QUICKSIGHT_OPERATIONAL_ISSUE_0CFEF_74AA9EB2514",
         "endTime":1.669662174E9,
         "eventScopeCode":"PUBLIC",
         "eventTypeCategory":"issue",
         "eventTypeCode":"AWS_QUICKSIGHT_OPERATIONAL_ISSUE",
         "lastUpdatedTime":1.669662173799E9,
         "region":"us-east-1",
         "service":"QUICKSIGHT",
         "startTime":1.669654485E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_SIMPLIFIED_AUTO_RECOVERY_SUCCESS/AWS_EC2_SIMPLIFIED_AUTO_RECOVERY_SUCCESS56c516c6-d39e-4b19-b4f3-29a334ebd3f2",
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"accountNotification",
         "eventTypeCode":"AWS_EC2_SIMPLIFIED_AUTO_RECOVERY_SUCCESS",
         "lastUpdatedTime":1.669567603257E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.66956760287E9,
         "statusCode":"open"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:us-east-1::event/EC2/AWS_EC2_OPERATIONAL_ISSUE/AWS_EC2_OPERATIONAL_ISSUE_0687F_2BF7DD21EBB",
         "endTime":1.669228246E9,
         "eventScopeCode":"PUBLIC",
         "eventTypeCategory":"issue",
         "eventTypeCode":"AWS_EC2_OPERATIONAL_ISSUE",
         "lastUpdatedTime":1.669289127226E9,
         "region":"us-east-1",
         "service":"EC2",
         "startTime":1.6692281E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED/AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED_d1d5ebd5-67c0-4d3d-afc1-5ed263ef2ebc",
         "endTime":1.668680309044E9,
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"scheduledChange",
         "eventTypeCode":"AWS_EC2_INSTANCE_REBOOT_MAINTENANCE_SCHEDULED",
         "lastUpdatedTime":1.668680324219E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.668680309044E9,
         "statusCode":"closed"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_ODCR_DEPOSIT_CONFIRMATION/AWS_EC2_ODCR_DEPOSIT_CONFIRMATION_239d634a-71ac-4af2-adc4-678b4f89463f",
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"accountNotification",
         "eventTypeCode":"AWS_EC2_ODCR_DEPOSIT_CONFIRMATION",
         "lastUpdatedTime":1.668150155089E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.668150154648E9,
         "statusCode":"open"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:eu-central-1::event/EC2/AWS_EC2_ODCR_DEPOSIT_CONFIRMATION/AWS_EC2_ODCR_DEPOSIT_CONFIRMATION_54abbe1b-772a-456a-9dbf-8aafb7547ccf",
         "eventScopeCode":"ACCOUNT_SPECIFIC",
         "eventTypeCategory":"accountNotification",
         "eventTypeCode":"AWS_EC2_ODCR_DEPOSIT_CONFIRMATION",
         "lastUpdatedTime":1.668150057236E9,
         "region":"eu-central-1",
         "service":"EC2",
         "startTime":1.668150056781E9,
         "statusCode":"open"
      },
      {
         "__type":"Event",
         "arn":"arn:aws:health:us-east-1::event/TRANSFER/AWS_TRANSFER_OPERATIONAL_ISSUE/AWS_TRANSFER_OPERATIONAL_ISSUE_2425A_18F932ED970",
         "endTime":1.667941291E9,
         "eventScopeCode":"PUBLIC",
         "eventTypeCategory":"issue",
         "eventTypeCode":"AWS_TRANSFER_OPERATIONAL_ISSUE",
         "lastUpdatedTime":1.667941291516E9,
         "region":"us-east-1",
         "service":"TRANSFER",
         "startTime":1.66793928E9,
         "statusCode":"closed"
      }
   ],
   "nextToken":
}
```
It seems that the response body doesn't contain the AZ information. 
If we want to get the AZ information for each event, do you have any suggestions?

**Comment by RanVaknin** [comment]
Hi @chrislinan ,

I'm not sure why it's missing from the response. I'm not aware of a different method that can return the desired AZ.
I was not able to find any documentation supporting this so I'm not sure if this is expected behavior. 

I have created an internal ticket with the team that owns that API. I can't commit to a timeline for the fix but I hope someone can get that looked at soon.

I will keep this issue open for tracking.
Thank you,
Ran

`P76910230`
